posthog/queries/event_query/event_query.py

```diff
@@ -105,6 +107,19 @@ def _get_distinct_id_query(self) -> str:
         else:
             return ""
 
+    def _get_person_id_overrides_query(self) -> str:
+        if self._should_join_distinct_ids:
+            return f"""
+            LEFT OUTER JOIN (
+                SELECT override_person_id as person_id, old_person_id
+                FROM person_overrides
+                WHERE team_id = %(team_id)s
+            ) AS overrides
+            ON {self.EVENT_TABLE_ALIAS}.person_id = overrides.old_person_id
```

Comment:
@PostHog/team-pipeline first thing to look at

Reply:
just so I'm understanding correctly:

1. Left outer join because not everything will have an override
2. Only runs when person on events is enabled (hence e.person_id exists)
3. Replaces the pdi join, hence only needs to happen when should_join_distinct_ids

Reply:
Correct me if I'm wrong, but it doesn't seem like `event_query` cares about whether it's v1 or v2.

Not making these separate seems nicer because otherwise we can end up actually joining the `personQuery` right now, because it only checks for v1 enabled 🤔